cmd/util/runtimeUlimits.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright © 2020-2022 The k3d Author(s)
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+package util
+
+import (
+	"strconv"
+	"strings"
+
+	dockerunits "github.com/docker/go-units"
+
+	"github.com/k3d-io/k3d/v5/pkg/config/v1alpha5"
+	l "github.com/k3d-io/k3d/v5/pkg/logger"
+)
+
+type UlimitTypes interface {
+	dockerunits.Ulimit | v1alpha5.Ulimit
+}
+
+type Ulimit[T UlimitTypes] struct {
+	Values T
+}
+
+// ValidateRuntimeUlimitKey validates a given ulimit key is valid
+func ValidateRuntimeUlimitKey(ulimitKey string) {
+	ulimitsKeys := map[string]bool{
```

Comment:
Can we get a list of supported keys from somewhere e.g. in the Docker SDK?
Would be nicer than maintaining it over here.
I'd be fine with it for a start though.

Reply:
Could be `interface{}` instead of bool, but I see the benefit of the latter a little further down

Reply:
I originally attempt to do so but the map var of it in the go-units is private 
https://github.com/docker/go-units/blob/master/ulimit.go#L46